src/Avalonia.X11/X11Window.cs

```diff
@@ -1146,6 +1150,35 @@ public void SetIcon(IWindowIconImpl icon)
             }
         }
 
+        /// <summary>
+        ///   Sets the application name that is visible for example in the Gnome desktop top bar
+        /// </summary>
+        /// <param name="name">The application name to set</param>
+        public void SetApplicationName(string name)
+        {
+            var classHint = XAllocClassHint();
+
+            if (classHint == IntPtr.Zero)
+            {
+                return;
+            }
+
+            var classHintData = Marshal.PtrToStructure<XClassHint>(classHint);
```

Comment:
It's better to just cast it to a pointer

Reply:
Sorry, I'm not super familiar with native interop in C# yet, what exactly do you mean? If I try code like `(XClassHint*)classHint` that doesn't compile with "Cannot declare a pointer to managed type". And the first result on StackOverflow said to just use `Marshal.PtrToStructure`.

Reply:
`(XClassHint*)classHint.ToPointer()`

Reply:
I had to modify my struct to get that to compile to:
```c#
        [StructLayout(LayoutKind.Sequential)]
        public struct XClassHint
        {
            public void* ResName;
            public void* ResClass;
        }
```